tst.py

```diff
@@ -1670,6 +1670,21 @@ class C:
         # Make sure we're still talking to our custom mapping.
         self.assertEqual(fields(C)['i'].metadata['xyzzy'], 'plugh')
 
+    def test_generic_dataclasses(self):
+        T = TypeVar('T')
+
+        @dataclass
+        class LabeledBox(Generic[T]):
+            content: T
+            label: str = '<unknown>'
+
+        box = LabeledBox(42)
+        self.assertEqual(box.content, 42)
+        self.assertEqual(box.label, '<unknown>')
+
+        # subscripting the resulting class should work, etc.
+        boxes: List[LabeledBox[int]] = []
+
```

Comment:
I would add more tests with subclassing/extending. For example:
```python
@dataclass
class Base(Generic[T, S]):
    x: T
    y: S

@dataclass
class DataDerived(Base[int, T]):
    new_field: str  # By the way, what do we do if a non-default field follows a default one
                    # in typing I think we raise a ValueError

Alias = DataDerived[str]

class NonDataDerived(Base[int, T]):
    def new_method(self): ...
```
and so on.

Reply:
Thanks.

As to the note about non-default fields following default fields: it raises a `TypeError`, even if the non-default field is added by inheritance. See the `test_two_fields_one_default` test.

Reply:
> it raises a ``TypeError``

(I double-checked and ``typing`` also raises ``TypeError``, so it is consistent.)